tools/chplcheck/src/chplcheck.py

```diff
@@ -282,8 +289,9 @@ def main():
                 printed_warning = True
 
     if printed_warning:
-        sys.exit(1)
+        return 1
 
 
 if __name__ == "__main__":
-    main()
+    ret = main()
+    sys.exit(ret if ret is not None else 0)
```

Comment:
You've adjusted all returns to have a numeric value, so you don't need this condition, do you?